app/services/manifest_submitter.rb

```diff
@@ -0,0 +1,26 @@
+class ManifestSubmitter
+  include Pundit
+  attr_reader :args
+
+  def initialize(args)
+    @args = args
+  end
+
+  def perform
+    output_stream = StreamLogger.new(Rails.logger)
+
+    manifest = Manifest.find_by_uuid_or_tracking_number!(args[:manifest_id])
+    authorize manifest, :can_submit?
```

Comment:
ah I see.

I would suggest putting the authorization check *outside* the service. Strong barriers at the castle walls mean the service can just do its one thing.